numpy/core/shape_base.py

```diff
@@ -356,3 +361,105 @@ def stack(arrays, axis=0):
     sl = (slice(None),) * axis + (_nx.newaxis,)
     expanded_arrays = [arr[sl] for arr in arrays]
     return _nx.concatenate(expanded_arrays, axis=axis)
+
+
+def block(*arrays):
+    """
+    Create a block array consisting of other arrays.
```

Comment:
Change to `Assemble arrays from blocks consisting of other arrays.` Without further explanation, the term "block array" does not have an unambiguous meaning.